src/components/TextInput/TextInput.test.jsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+import {
+  render,
+  fireEvent,
+  screen,
+  waitFor,
+} from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import TextInput from './TextInput';
+
+describe('TextInput', () => {
+  /**
+   * Props Validation
+   */
+  test('Throws error if required prop "value" is not supplied to component', () => {
+    console.error = jest.fn(); // eslint-disable-line no-console
+    render(<TextInput onChange={() => null} />);
+    expect(console.error).toHaveBeenCalledTimes(1); // eslint-disable-line no-console
+    expect(console.error.mock.calls[0][0]) // eslint-disable-line no-console
+      .toContain('Failed prop type: The prop `value`');
+  });
+
+  test('Throws error if required prop "onChange" is not supplied to component', () => {
+    console.error = jest.fn(); // eslint-disable-line no-console
+    render(<TextInput value="hello" />);
+    expect(console.error).toHaveBeenCalledTimes(1); // eslint-disable-line no-console
+    expect(console.error.mock.calls[0][0]) // eslint-disable-line no-console
+      .toContain('Failed prop type: The prop `onChange`');
+  });
+
+  test('Throws an error if "type" prop is anything other than allowed values', () => {
+    console.error = jest.fn(); // eslint-disable-line no-console
+    render(<TextInput onChange={() => null} value="hello" type="notOnTheList" />);
+    expect(console.error).toHaveBeenCalledTimes(1); // eslint-disable-line no-console
+    expect(console.error.mock.calls[0][0]) // eslint-disable-line no-console
+      .toContain('Failed prop type: Invalid prop `type`');
+  });
```

Comment:
we actually dont need these validation texts when we change this to typescript as the code wont even compile if the necessary props aren't passed in

Reply:
Hey @ericyliu perhaps I'm misunderstanding, but the goal of these is to guard against somebody flipping a prop requirement. Basically this test breaks to let you know that the prop was required for a reason so best not to remove that requirement without explicit intent. Let me know if that makes sense.

Reply:
ohh good point, yeah that's reasonable